client/post-editor/controller.js

```diff
@@ -247,7 +247,17 @@ export default {
 				startEditingPostCopy( site, postToCopyId, context );
 			} else if ( postID ) {
 				// TODO: REDUX - remove flux actions when whole post-editor is reduxified
-				actions.startEditingExisting( site, postID );
+				actions.startEditingExisting( site, postID ).then( editedPost => {
+					if ( context.path !== page.current ) {
```

Comment:
to be ultra-safe here, we might want to stash a direct reference to the current path instead of accessing it via `context`. There's a chance that someone else changes the prop between the time we register this callback and the time the api call completes. So

```
const currentPath = context.path;
actions.startEditingExisting( ... ).then(... => if ( currentPath !== page.current ) )
```

Reply:
Good idea, let's do it.